packages/palette/src/elements/Checkbox/Checkbox.tsx

```diff
@@ -33,144 +18,78 @@ export interface CheckboxProps {
   onSelect?: (selected: boolean) => void
 }
 
-export interface CheckboxToggleProps
-  extends CheckboxProps,
-    BorderProps,
-    SizeProps,
-    SpaceProps {}
-
-/**
- * A checkbox
- */
-export class Checkbox extends React.Component<CheckboxProps> {
-  labelColor = () => {
-    const { disabled, error } = this.props
-    if (disabled) return { color: color("black10") }
-    if (error) return { color: color("red100") }
-  }
-
-  iconColor = () => {
-    const { disabled, selected } = this.props
-    if (disabled && selected) return "black30"
-    if (disabled) return "black5"
-    return "white100"
-  }
-
-  render() {
-    const {
-      selected,
-      children,
-      error,
-      disabled,
-      hover,
-      onSelect,
-      ...other
-    } = this.props
-
-    return (
-      <Container
-        className={hover && "hover"}
-        my={0.5}
-        onClick={() => !disabled && onSelect && onSelect(!selected)}
-        selected={selected}
-        hover={hover}
-        disabled={disabled}
-        alignItems="center"
-        {...other}
-      >
-        <CheckboxButton
-          border={1}
-          mr={1}
-          selected={selected}
-          error={error}
-          disabled={disabled}
-        >
-          <CheckIcon fill={this.iconColor()} />
-        </CheckboxButton>
-        <Label style={this.labelColor()}>{children}</Label>
-      </Container>
-    )
-  }
-}
-
-interface CheckboxFeedbackState {
-  disabled?: boolean
-  selected?: boolean
-  error?: boolean
-}
-
-const checkBorderColor = ({
-  disabled,
+/** A checkbox */
+export const Checkbox: React.FC<CheckboxProps> = ({
+  className,
   selected,
+  children,
   error,
-}: CheckboxFeedbackState) => {
-  if (disabled) return color("black10")
-  if (selected) return color("black100")
-  if (error) return color("red100")
-  return color("black10")
-}
-
-const checkBackgroundColor = ({
   disabled,
-  selected,
-}: CheckboxFeedbackState) => {
-  switch (true) {
-    case disabled:
-      return color("black5")
-    case selected:
-      return color("black100")
-    default:
-      return color("white100")
+  hover,
+  onSelect,
+  onClick,
+  ...rest
+}) => {
+  const labelColor = () => {
+    if (disabled) return "black10"
+    if (error) return "red100"
   }
-}
 
-const CheckboxButton = styled.div<CheckboxToggleProps>`
-  ${borders};
-  ${styledSpace};
-  background-color: ${checkBackgroundColor};
-  border-color: ${checkBorderColor};
-  width: ${space(SIZE)}px;
-  height: ${space(SIZE)}px;
-  transition: background-color 0.25s, border-color 0.25s;
-
-  svg {
-    position: relative;
-    top: -${BORDER_WIDTH}px;
-    left: -${BORDER_WIDTH}px;
+  const handleClick = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ) => {
+    !disabled && onSelect && onSelect(!selected)
+    onClick && onClick(event)
```

Comment:
Totally non blocking... long term across a wide dev team, conventional `if`s are more readable, less bug prone, less prone to needing to be unwound (like implicityly returned react components), more "boring" (better), etc.